### PR TITLE
Add docs 

### PR DESCRIPTION
**Pull request by theFeu**

### CODE REVIEW

**Comment by theFeu** [comment]
Needs Review :)

**Comment by dnsmichi** [comment]
Notes from a first look:

- Needs a rebase against current master
- IcingaDB itself is not listed as dependency
- Unclear naming - `IcingaDB web module` and later `Icinga DB`
- Installation is too short, this should align with detailed instructions for downloading and extracting the module, like recently done for Graphite. https://github.com/Icinga/icingaweb2-module-graphite/blob/master/doc/02-Installation.md#install-the-graphite-module

I'll continue with inline remarks.

File: doc/03-Configuration.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Option             | Description
+-------------------|------------------------
+Redis Host         | IP adress or FQDN, usually this is your Icinga Master host
+Redis Port         | The port over which you can connect to the Redis
+
+## Command Transports
+
+Set command transports for specific Icinga instances.  
+For more information check out the [documentation](https://github.com/Icinga/icingaweb2/blob/master/modules/monitoring/doc/05-Command-Transports.md) in the monitoring module.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: This should link to https://icinga.com/docs/icingaweb2/latest/modules/monitoring/doc/05-Command-Transports/ ... the same applies to any other online reference, we decided to try to move users solely to `icinga.com/docs` being the only source of truth.

File: doc/03-Configuration.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+Option             | Description
+-------------------|------------------------
+Redis Host         | IP adress or FQDN, usually this is your Icinga Master host
+Redis Port         | The port over which you can connect to the Redis
+
+## Command Transports
+
+Set command transports for specific Icinga instances.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: A more detailed introduction would help, at the very moment starting fresh with Icinga I have no idea about this term.

File: doc/03-Configuration.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+Redis Port         | The port over which you can connect to the Redis
+
+## Command Transports
+
+Set command transports for specific Icinga instances.  
+For more information check out the [documentation](https://github.com/Icinga/icingaweb2/blob/master/modules/monitoring/doc/05-Command-Transports.md) in the monitoring module.
+
+## Security
+
+Protect your custom variables with a comma separated, case insensitive list that supports wildcards.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: - Link or explain what a custom variable is
- Provide an example which includes multiple items (comma), case sensitivity, and also wildcards

File: doc/03-Configuration.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+This is how you configure your Icinga DB so it works right off the bat.
+
+## Database
+
+Select the database resource for your setup.
+
+## Redis
+
+Configure the Redis connections details of your Icinga host.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: - s/connections/connection/
- Explain or link where the Redis component is described in detail. Which service or package is responsible for setting this up?

File: doc/03-Configuration.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+This is how you configure your Icinga DB so it works right off the bat.
+
+## Database
+
+Select the database resource for your setup.
+
+## Redis
+
+Configure the Redis connections details of your Icinga host.  
+If you are running a master-master setup, the first 2 fields are for your configuration master.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: - Which fields are you referring to? (missing screenshot)
- What exactly is a "master-master" setup? This terminology doesn't exist in the Icinga 2 distributed monitoring docs. Better use something like "a high availability zone with two masters".

File: doc/03-Configuration.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+
+Option             | Description
+-------------------|------------------------
+Redis Host         | IP adress or FQDN, usually this is your Icinga Master host
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: - s/adress/address/

File: doc/03-Configuration.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+Redis Port         | The port over which you can connect to the Redis
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: `to the Redis` ... either `to Redis` or yet better, `The port where the Redis server is listening on`.

File: doc/11-Concepts.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
+# Views
+
+## List Items
+The lists are one of the most prominent views for the monitoring. In the default view they follow this style:
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Bold statement, why are lists one of the most prominent views?

File: doc/03-Configuration.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# Configuration
+
+This is how you configure your Icinga DB so it works right off the bat.
+
+## Database
+
+Select the database resource for your setup.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: I would assume that I have to create the DB as a resource first, right? Please explain that to me, or show a screenshot where to find this and its credentials. Also, add a table with the attributes like for Redis.

File: doc/02-Installation.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+* Icinga Web 2 modules:
+  * [reactbundle](https://github.com/Icinga/icingaweb2-module-reactbundle) (>= 0.4) (Icinga Web 2 module)
+  * [Icinga PHP Library (ipl)](https://github.com/Icinga/icingaweb2-module-ipl) (>= 0.5) (Icinga Web 2 module)
+
+## Installation
+
+1. Just drop this module to a `icingadb` subfolder in your Icinga Web 2 module path.
+
+2. Log in with a privileged user in Icinga Web 2 and enable the module in `Configuration -> Modules -> icingadb`.
+Or use the `icingacli` and run `icingacli module enable icingadb`.
```
Comment (dnsmichi) [inline doc/02-Installation.md]: Replace 1. and 2. with the portions of https://github.com/Icinga/icingaweb2-module-graphite/blob/master/doc/02-Installation.md#install-the-graphite-module

File: doc/02-Installation.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# <a id="Installation"></a>Installation
+
+## Requirements
+
+* Icinga Web 2 (&gt;= 2.8 RC1)
+* PHP (&gt;= 5.6, preferably 7.x)
```
Comment (dnsmichi) [inline doc/02-Installation.md]: IIRC a new dependency is php-redis, or is that provided somewhere else?

For RHEL8, it seems that php-pecl-redis is not yet packaged.
https://serverfault.com/questions/997028/where-is-the-redis-module-for-php-on-centos-8-php-pecl-redis

File: doc/02-Installation.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+* Icinga Web 2 modules:
+  * [reactbundle](https://github.com/Icinga/icingaweb2-module-reactbundle) (>= 0.4) (Icinga Web 2 module)
+  * [Icinga PHP Library (ipl)](https://github.com/Icinga/icingaweb2-module-ipl) (>= 0.5) (Icinga Web 2 module)
```
Comment (dnsmichi) [inline doc/02-Installation.md]: Just a remark: That release doesn't yet exist as it needs to bundle everything.

File: README.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+## <a id="About"></a>
+
+The Icinga DB web module is the way to go, to interact with the new database and to use the benefits it brings.
+
+This module reimplements and aims to replace the old monitoring module.
+It keeps all functionality from its predecessor and adds new quality of life features, such as:
+- bulk operations: set entire host groups into downtime with one operation
+- modals instead of tabs to speed up the workflow
+- a view switcher, that enables the user to choose how much detail should be displayed
+- the status bar now checks for set filters
```
Comment (dnsmichi) [inline README.md]: I would replace the above paragraph entirely and avoid comparisons with the old module where applicable. It should be catchy and encouraging to try things immediately. The "About" section is where users read and continue, or close the page.


```
The Icinga DB web module pushes your monitoring stack to the next level. 

Based on the lessons learnt with the base monitoring module, it offers a modern and streamlined design to provide a clear and concise overview on your monitoring environment.

The module adds new features, such as:

- Bulk operations to schedule downtimes for host groups and alike.
- Enhanced action workflows with In-window modals 
- View switching to chose how much detail will be displayed
- Improved status bar with applied filters
```

File: README.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+On top it offers a modern and streamlined design to give an even clearer and more concise overview over the most important points in your environment.
```
Comment (dnsmichi) [inline README.md]: Oh, this line is also meant with the `above paragraph`.

File: README.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+![ListTypes](doc/res/ListTypes.jpg "Different levels of list details")
+
+![HostServiceList](doc/res/ScreenHostServiceList.jpg "Host and Service list")
+
+![HostDetailHistory](doc/res/ScreenHostDetailHistory.jpg "Host deatil and history")
+
+## <a id="Documentation"></a>Documentation
+
+The full documentation is located in the [doc/](doc/) directory
```
Comment (dnsmichi) [inline README.md]: Remove the word `full`, it is redundant.

Also, this link won't work on icinga.com/docs. Better link to `https://icinga.com/docs/<futureprojectname>`

File: README.md [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+
+## <a id="Installation"></a>Installation
+
+For installing Icinga DB please check [installation](doc/02-Installation.md)
```
Comment (dnsmichi) [inline README.md]: Refine the wording:  `check the [installation](doc/02-Installation.md) chapter`.

File: README.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+in the documentation.
+
+## <a id="Configuration"></a>Configuration
+
+To configure your Icinga DB please check [configuration](doc/03-Configuration.md).
```
Comment (dnsmichi) [inline README.md]: Refine wording: `check the [configuration](doc/03-Configuration.md) chapter`.

File: README.md [hunk 18]
```diff
@@ -0,0 +1,39 @@
+# Icinga DB
+
+1. [About](#about)
+2. [Documentation](#documentation)
+3. [Installation](#installation)
+4. [Configuration](#configuration)
```
Comment (dnsmichi) [inline README.md]: Please add absolute anchors including the filename here. This avoids possible failures.

Example:
```
[About](README.md#about)
```

File: doc/01-About.md [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+
+This module reimplements and aims to replace the old monitoring module.
+It keeps all functionality from its predecessor and adds new quality of life features, such as:
+- bulk operations: set entire host groups into downtime with one operation
+- modals instead of tabs to speed up the workflow
+- a view switcher, that enables the user to choose how much detail should be displayed
+- the status bar now checks for set filters
+
+On top it offers a modern and streamlined design to give an even clearer and more concise overview over the most important points in your environment.
+
```
Comment (dnsmichi) [inline doc/01-About.md]: Once https://github.com/Icinga/icingaweb2-module-icingadb/pull/1/files#r366814795 is fixed, please change its content here as well.

File: doc/02-Installation.md [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+
+## Installation
+
+1. Just drop this module to a `icingadb` subfolder in your Icinga Web 2 module path.
+
+2. Log in with a privileged user in Icinga Web 2 and enable the module in `Configuration -> Modules -> icingadb`.
+Or use the `icingacli` and run `icingacli module enable icingadb`.
+
+This concludes the installation. You should now be able to use the Icinga DB module.
```
Comment (dnsmichi) [inline doc/02-Installation.md]: We need to configure it first, right? 

Better would be something like: `Open the Icinga Web dashboard. If you need to configure additional settings, please check the [configuration](doc/03-Configuration.md) chapter.`

File: doc/03-Configuration.md [hunk 21]
```diff
@@ -0,0 +1,28 @@
+# Configuration
+
+This is how you configure your Icinga DB so it works right off the bat.
```
Comment (dnsmichi) [inline doc/03-Configuration.md]: We're configuring the `Icinga DB web module` here. Also, it should mention that it comes with default settings.

File: doc/11-Concepts.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+![DefaultAnatomy](res/ListItemAnatomy.jpg "A list item stripped down to the skeleton")
+
+### Visual
+Each element has a so-called visual. It's purpose is to highlight certain elements in long lists and give an intuitive overview about the underlying objects' condition. 
+In the case of host or service lists, for example, the state is displayed. This makes it immediately clear if and where there are problems that need to be taken care of, immediately from the overview.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Minor nitpick: Remove `for example`.

`immediately` is repeated in the sentence, find a better wording.

File: doc/11-Concepts.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+
+### Title
+The title briefly describes the state of the list element as an addition to the visual. For example, it contains the information that a host is currently DOWN. While the visual gives an intuitive impression, the title explains what exactly happened.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: `that a host is currently in the DOWN state`.

File: doc/11-Concepts.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+
+### Meta
+The meta area is for additional information, usually displays of time. In the Host and Service Lists, you can see how long the object has been in the corresponding state.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Missing verb, maybe better `The meta area is intended to provide additional details, usually displays of time.`

What exactly is a `corresponding state`? Maybe link to the Icinga 2 docs, https://icinga.com/docs/icinga2/latest/doc/03-monitoring-basics/#host-states

Comment (dnsmichi) [reply doc/11-Concepts.md]: Besides, `Host and Service Lists` ... why capitalized?

File: doc/11-Concepts.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+
+### Caption
+The caption area contains detailed information about the list item. In the case of hosts and services this would be the plugin output. For comments and downtimes the comment texts of the user are displayed here. To make the default list view compact and consistent, long texts are truncated to a maximum of two lines.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Replace `plugin output` with `check output`. This aligns with `check source` and everything else. Icinga's check results are not necessarily populated from plugins, and new users may not understand the term at this point.

File: doc/11-Concepts.md [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+
+## Different detail degrees
+
+The level of detail of the host and service lists is now selectable - the above image shows the default list item, but there are two more iterations:
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Remove `now`, we shall not compare unless explicitly mentioned.

File: doc/11-Concepts.md [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+In the compact mode the entire list item is displayed in one single line.
+Given, that there is enough space, the caption is placed in the same line as the rest and trimmed to fit.
+![CompactAnatomy](res/ListItemAnatomyCompact.jpg "A more compact list item that fits in one line")
+
+The detailed mode displays the entire caption, which makes it possible to react to problems directly from the list view.
+![DetailedAnatomy](res/ListItemAnatomyDetailed.jpg "A more detailed list item that spans four lines")
+
+## Overdue Items
+
+In host and service lists overdue checks are now highlighted. This makes it immediately apparent which objects may no longer be up-to-date. The red badge contains information about how long the check has overdue.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Remove `now`.

File: doc/11-Concepts.md [hunk 28]
```diff
@@ -0,0 +1,77 @@
...
+![OverdueListAnatomy](res/ListAnatomyOverdue.jpg "A list with a highlighted item")
+
+## Downtimes
+Downtime lists have an extra information to be displayed: the progress.
+How far along the downtime is, is now immeadiately visible via the progress bar at the top of each list item.
+In the only square visual, we display how much time is left until the downtime ends.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Replace `we`, no impersonation needed from the overall docs style.

File: doc/11-Concepts.md [hunk 29]
```diff
@@ -0,0 +1,77 @@
...
+![DowntimeListItem](res/ListItemDowntimeAnatomy.jpg "A downtime list item with the added progress bar in the top")
+
+
+# Modal
+If you wanted to create a comment or set a downtime from the detail area, the dialog was displayed in another column, which meant that the contents of the left column were lost. In the icingadb module, there is now a modal element for short interaction dialogues. For actions in the detail area, a modal dialog will now appear. This preserves the left list column and thus the context better.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Remove the first sentence as it compares to something unmentioned. 

Start with 

```
The modal element introduces short interaction dialogues. For actions in the detail area, a modal dialog will appear. This preserves the left list column and thus the content better.
```

Also, avoid `now` again.

File: doc/11-Concepts.md [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+![ModalAnatomy](res/ModalAnatomy.jpg "A modal presenting a form")
+
+### Utility
+The utility or close button is one if the three ways to close the modal without confirming your changes. You may also just click outside it's borders or press the escape key to escape its' grasp.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - s/is one if the three/is one of three/ ... Fix typo and avoid the `the` making this absolute.
- Remove `your` .. from now on the impersonation changes in this document. 
- s/it's/its/
- I don't understand the intention behind `press the escape key to escape its' grasp.`.

File: doc/11-Concepts.md [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+
+### Title
+The title tells you what is asked off you in a simple keyword. Usually an action like: Acknowledge Problem, Add Comment or Schedule Downtime.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - You again.
- What is asked off me ... I don't understand that sentence. Since it refers to an action later, why not rephrase that to `The title specifies the target, e.g. an action like: ...`

File: doc/11-Concepts.md [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+
+### Hint
+The hint explains more about how to interact with the modal and what is asked of you. It will also go into detail what consequences your actions within the modal will have on your environment.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - you, your.
- Better: `It goes into detail about the consequences from an action to your environment`

File: doc/11-Concepts.md [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+
+### Form
+The main reason for modals is to perform an action without getting a new tab with your form, which would switch the layout you were working with before.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - your, you, especially since it is not my form in this case.

File: doc/11-Concepts.md [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+
+### Submit
+With the submit button you apply the choses you made in the form and return to where you left off.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - you
- `choses` ? I'd change that to `selection` or `changes`.

File: doc/11-Concepts.md [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+
+# Details
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: New line below.

File: doc/11-Concepts.md [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+## Check statistics
+Information about check execution is available in a handy graph that makes it possible to get all of the information at a glance.
+![CheckStatistics](res/CheckStatisticsAnatomy.jpg "A panel with the statistics about check execution")
+
+### Descriptions and Values
+Instead of having a table with the different check information there is now a header that lays out the information in an intuitive way, giving more weight to more important bits of data and their values.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - Remove `now`.
- Remove the second `more`.

File: doc/11-Concepts.md [hunk 37]
```diff
@@ -0,0 +1,77 @@
...
+
+### Graph
+The graphical element in the middle makes it easier to see, whether the check is on time or late. It's easy to immediately see the configured check intervall and have a better grasp on how up-to-date the current state is.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: - Which `middle` is referred to?
- s/intervall/interval/g

File: doc/11-Concepts.md [hunk 38]
```diff
@@ -0,0 +1,77 @@
...
+
+### Indicator
+The indicator visualises where the "now" is for orientation.
```
Comment (dnsmichi) [inline doc/11-Concepts.md]: Maybe rephrase this to `The indicator marks the current date and time for better orientation.`.

File: doc/02-Installation.md [hunk 39]
```diff
@@ -0,0 +1,18 @@
+# <a id="Installation"></a>Installation
+
+## Requirements
+
```
Comment (dnsmichi) [inline doc/02-Installation.md]: It should mention that you need Icinga DB itself, and link to it.

**Comment by dnsmichi** [comment]
# Summary

I did not test the installation, just followed the documentation from the flow. I'm playing the dumb user card as well as the professional who has heard about Icinga DB at OSMC. Last but not least, marketing and promotion.

- About/Introduction needs more catchy and candy phrasing. Avoid comparing with the old module too much, not everyone knows how this works. Highlight and illustrate the IcingaDB module's bullet proof features.
- Installation needs to be enhanced with copy paste instructions for everyone to try
  - Add the IcingaDB project dependency, php-redis
- Configuration needs the database resource description
- The concepts chapter changes into "we and you" impersonation at some point. This is irritating when reading and should be generally avoided, since documentation shouldn't do that.
- Some typos, wrong phrases, missing verbs
- Naming, this is the Icinga DB Web module 
- Some details are taking as granted, while you cannot expect that from the user. Wherever applicable, link to existing icinga.com/docs resources
- Ensure that users have an easy-to-install experience from following the docs. Highlight dependencies, such as PHP and Redis explicitly. 

@nilmerg @lippserd handing over to you again :)

File: README.md (hunk 13 above)
Comment (theFeu) [reply README.md]: Changed, thanks for the suggestion!

File: README.md (hunk 15 above)
Comment (theFeu) [reply README.md]: Removed the `full`
Not sure about the project name - so I'll leave this open for now :)

File: README.md (hunk 16 above)
Comment (theFeu) [reply README.md]: Gotcha!

File: README.md (hunk 18 above)
Comment (theFeu) [reply README.md]: For me this opens a new tab instead of jumping in the file - not sure if that is the intended effect?

If it is, we need to change that in the icingaweb2 README as well.

File: doc/01-About.md (hunk 19 above)
Comment (theFeu) [reply doc/01-About.md]: aye!

File: doc/02-Installation.md (hunk 20 above)
Comment (theFeu) [reply doc/02-Installation.md]: Added hint about configuration

File: doc/03-Configuration.md (hunk 21 above)
Comment (theFeu) [reply doc/03-Configuration.md]: makes sense!

File: doc/11-Concepts.md (hunk 24 above)
Comment (theFeu) [reply doc/11-Concepts.md]: Reworded and removed the capitalisation

File: doc/11-Concepts.md (hunk 25 above)
Comment (theFeu) [reply doc/11-Concepts.md]: good point, thanks!

File: doc/11-Concepts.md (hunk 30 above)
Comment (theFeu) [reply doc/11-Concepts.md]: Oh my, that paragraph was a mess... 🙈

**Comment by dnsmichi** [comment]
Some parts require linking to IcingaDB where this PR needs to be reviewed/merged - please coordinate this: https://github.com/Icinga/icingadb/pull/3 - this then needs https://icinga.com/docs/latest/icingadb as a project on the web host, being able to link to. Once the PR is merged, ask @bobapple for the details please.

**Comment by theFeu** [comment]
All of the semantics have been fixed - all that remains is more technical, and I think it might be easier if one of the IcingaDB devs handle the rest? :)

**Comment by theFeu** [comment]
Rebase with master done as well, back to you @lippserd :)

**Comment by dnsmichi** [comment]
@theFeu seems the rebase to master was incomplete, there are lots of commits not belonging here.

**Comment by theFeu** [comment]
> @theFeu seems the rebase to master was incomplete, there are lots of commits not belonging here.

However that happened, should be fixed now.
Thanks for pointing it out!
